modules/backup-gcs/client.go

```diff
@@ -216,32 +182,32 @@ func (g *gcsClient) PutObject(ctx context.Context, backupID, key string, byes []
 	return nil
 }
 
-func (g *gcsClient) Initialize(ctx context.Context, backupID string) error {
+func (g *gcsClient) Initialize(ctx context.Context, backupID, overrideBucket, overridePath string) error {
 	key := "access-check"
 
-	if err := g.PutObject(ctx, backupID, key, []byte("")); err != nil {
-		return errors.Wrap(err, "failed to access-check gcs backup module")
+	if err := g.PutObject(ctx, backupID, key, overrideBucket, overridePath, []byte("")); err != nil {
+		return errors.Wrapf(err, "failed to access-check gcs backup module %v %v %v %v", overrideBucket, overridePath, backupID, key)
 	}
 
-	bucket, err := g.findBucket(ctx)
+	bucket, err := g.findBucket(ctx, overrideBucket)
 	if err != nil {
 		return errors.Wrap(err, "find bucket")
 	}
 
-	objectName := g.makeObjectName(backupID, key)
+	objectName := g.makeObjectName(overridePath, []string{backupID, key})
 	if err := bucket.Object(objectName).Delete(ctx); err != nil {
-		return errors.Wrap(err, "failed to remove access-check gcs backup module")
+		return errors.Wrapf(err, "failed to remove access-check gcs backup module %v", objectName)
 	}
 
 	return nil
 }
 
 // WriteToFile downloads an object and store its content in destPath
 // The file destPath will be created if it doesn't exit
-func (g *gcsClient) WriteToFile(ctx context.Context, backupID, key, destPath string) (err error) {
-	bucket, err := g.findBucket(ctx)
+func (g *gcsClient) WriteToFile(ctx context.Context, backupID, key, destPath, overrideBucket, overridePath string) (err error) {
+	bucket, err := g.findBucket(ctx, overrideBucket)
 	if err != nil {
-		return fmt.Errorf("find bucket: %w", err)
+		return fmt.Errorf("WriteToFile: %w", err)
```

Comment:
suggestion: maybe writetofile: find bucket: %w error message would be more descriptive in this case?

Reply:
done